### PR TITLE
Feature/19 central screen header

### LINKED ISSUE #19: Implement proper screen header
**Describe the solution you'd like**
Implement a proper screen header that uses React Native Navigation.
List of features include:

- Title (center)
- Avatar (right)
  - clickable, callback can be provided
  - Component must be provided
- Action Button
  - Component must be provided
  - Can for example be a share button, or "Save" button with an icon
- Back button must stay working as is from react navigation

**Additional context**
-

### PR DESCRIPTION
**Pull request by alexw-at**
Closes #19 
This PR adds a customizable screen header.
- Screen Title is always required
- Screen Action is optional
- User Avatar is optional

### CODE REVIEW

File: src/components/ScreenHeader.tsx [hunk 1]
```diff
@@ -1,42 +1,88 @@
...
+                containerStyle={styles.actionButton}
+            />
+            {props.user !== undefined && (
+                <TouchableOpacity
+                    style={{ marginLeft: "auto" }}
+                    onPress={() =>
+                        console.log("press avatar of user: " + props.user?.id)
+                    }
+                >
                     <Avatar
```
Comment (valentingc) [inline src/components/ScreenHeader.tsx:66]: At least at my iPhone, the Avatar is cropped off (too big) - so I think I think the header should be a bit higher

File: src/screens/Itinerary/TripItinerary.tsx [hunk 2]
```diff
@@ -43,6 +44,29 @@ export default function TripItinerary(props: Props): JSX.Element {
         props.navigation.navigate(Routes.ADD_EDIT_ACTIVITY_GROUP, routeOpts);
         console.log("Add activity group button pressed");
     };
+
+    props.navigation.setOptions({
+        headerBackTitleVisible: false,
+        // eslint-disable-next-line react/display-name
+        headerTitle: (props2) => (
```
Comment (valentingc) [inline src/screens/Itinerary/TripItinerary.tsx]: I think props2 is a bit bad naming, headerProps or so would maybe be better I think
```suggestion
        headerTitle: (headerProps) => (
```

File: src/screens/Itinerary/TripItinerary.tsx [hunk 3]
```diff
@@ -43,6 +44,29 @@ export default function TripItinerary(props: Props): JSX.Element {
...
+                    <Icon
+                        style={styles.iconButton}
+                        name="share"
+                        size={20}
+                        color="#222"
+                        type="font-awesome-5"
+                    />
+                }
+                actionCallback={() => console.log("share trip")}
+                {...props2}
```
Comment (valentingc) [inline src/screens/Itinerary/TripItinerary.tsx]: see above
```suggestion
                {...headerProps}
```

**Review by valentingc — CHANGES_REQUESTED** [review]
Looks good, just the header is a bit too small on my device, so the avatar is cropped off. Is it different for your device?

File: src/components/ScreenHeader.tsx (hunk 1 above)
Comment (alexw-at) [reply src/components/ScreenHeader.tsx:66]: The avatar is not cropped off on Android. I made it smaller to fit the iOS header.

**Review by valentingc — APPROVED** [review]

**Pull request merged**